### ISSUE TITLE
Metadata not written (and new visualizations not displaying) in 1.7.0-rc1

### ISSUE THREAD

**Issue by vaskozl** [kubeflow/kubeflow#6063]
/kind bug

I Kubeflow upgraded from 1.4.0 to 1.7.0-rc1 with the platnform-agnostic manifests.

While I now see correct visualizations of statistics from runs that happened before upgrading to 1.7.0-rc1, new runs only display the markdown details.

The TFX pipelines I submit are exactly the same. On the new runs the ML Metadata tab of the components prints:

"Corresponding ML Metadata not found."

**Anything else you would like to add:**

Furthermore I don't see any new executions on the executions page despite running many pipelines since upgrading.

I don't see anything special in the logs of the TFX pods except:
```
WARNING:absl:metadata_connection_config is not provided by IR.
```

But that was present before upgrading to 1.7.0-rc1.

The only errors I see in the `metadata-grpc-deployment` pod is:

```2021-07-22 12:29:15.910133: W ml_metadata/metadata_store/metadata_store_service_impl.cc:602] PutContexts failed: Given node already exists: type_id: 48
name: "sp-lstm-rh6xt"
Internal: mysql_query failed: errno: 1062, error: Duplicate entry '48-sp-lstm-rh6xt' for key 'type_id'
	Cannot create node for type_id: 48 name: "sp-lstm-rh6xt"
```

Basically I don't think executions and artifacts are getting written to the DB for some reason. Not sure how to debug this. This causes the visualizations to not show up as far as I can see.

Metadata in the TFX pipelines is configured via the `get_default_kubeflow_metadata_config` `tfx.orchestration.kubeflow` function.

**Environment:**

- Kubeflow version: 1.4.0 -> 1.7.0-rc1
- kfctl version: Not used. Using tfx.orchestration.kubeflow to submit pipelines.
- Kubernetes platform: Upstream kubeadm: k8s v1.20.5
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`): Centos 8

**Comment by vaskozl** [comment]
Closing in favour of issues in the pipelines repo: https://github.com/kubeflow/pipelines/issues/6138

**Comment by vaskozl** [comment]
I've also just tested the issues with 1.7.0-rc2.

I've tried kfp 1.4.0 and 1.6.6
